src/storage/btree_load.c

```diff
@@ -158,26 +158,38 @@ struct index_builder_loader_context
   std::atomic<std::uint64_t> m_tasks_executed;
   int m_tran_index;
   int m_error_code;
+  const TP_DOMAIN* m_key_type;
+};
+
+struct index_builder_key_oid
+{
+  DB_VALUE m_key;
+  OID m_oid;
 };
 
 class index_builder_loader_task: public cubthread::entry_task
 {
-private:
+  private:
     BTID m_btid;
-    DB_VALUE m_key;
-    OID m_class_oid, m_oid;
+    std::vector<index_builder_key_oid> m_keys_oids;
+    OID m_class_oid;
     int m_unique_pk;
     index_builder_loader_context & m_load_context; // Loader context.
+    size_t m_memsize;
 
-public:
-    index_builder_loader_task (const BTID * btid, const OID * class_oid, const OID * oid, int unique_pk,
+  public:
+    index_builder_loader_task (const BTID * btid, const OID * class_oid, int unique_pk,
                                index_builder_loader_context & load_context);
-    
-    void set_key (const DB_VALUE * key);
-
     ~index_builder_loader_task ();
+    
+    // add key to key set and return true if task is ready for execution, false otherwise
+    bool set_key (const DB_VALUE * key, const OID& oid);
```

Comment:
rename 'set_key' -> 'add_key'
if return type is an enum of 'continue_task, full_task' the code would be more readable.